### ISSUE TITLE
Use the releative path or the end subfolder name as the terminal label or title

### ISSUE THREAD

**Issue by maxrks** [microsoft/vscode#150626]
When integrated terminal opened from the context menu, would **the relative path or the subfolder name** be placed **as the label
of the termnial**, other than renaming by right clicking to type.

more, if current path in the termnial changed, could the termnial label or tile be changed automatically by the same.

![Snipaste_2022-05-29_12-12-52](https://user-images.githubusercontent.com/20557631/170852003-12d9cd69-5b68-459e-9cd1-18a10f506a95.jpg)
![Snipaste_2022-05-29_12-13-36](https://user-images.githubusercontent.com/20557631/170852005-8fcf4c87-89e6-4478-ba95-ab12406b62e5.jpg)

**Comment by meganrogge** [comment]
This is working for me. @Tyriar does it work on windows for you?

**Comment by maxrks** [comment]
It does with MacOS release but not on Windows.  **@meganrogge** 

MacOS(Intel)
![Snipaste_2022-06-01_07-33-14](https://user-images.githubusercontent.com/20557631/171300718-5ef43085-1a7d-4f78-879e-865ac6b4975c.jpg)

Windows
![Snipaste_2022-06-01_07-31-34](https://user-images.githubusercontent.com/20557631/171300713-aaad9989-2a3c-47ac-bfc1-b6d06ed6d6dd.jpg)

**Comment by maxrks** [comment]
If the space occupied by labels be splited enough and the full relateive path  could be shown instead of the subfolder name, I thought that would be better. Perhaps the relative path might be too long to appear as label.

**Comment by Tyriar** [comment]
This works when shell integration is enabled but not when it's disabled. This is because it's not possible to extract the cwd in a reliable way on Windows without some form of shell integration. You can try out the feature here which should make it work in pwsh:

```
"terminal.integrated.shellIntegration.enabled": true
```

**Comment by maxrks** [comment]
Thanks! it does work with pwsh but if the terminal be changed with **Comannd Prompt** or **Git-Bash shell**, it shall fail.  **@Tyriar**
![Snipaste_2022-06-03_16-29-43](https://user-images.githubusercontent.com/20557631/171808851-a2b61aa9-76c2-4cf7-874b-0ad9ea70dd54.png)

**Comment by Tyriar** [comment]
Yes we only support shell integration in pwsh for Windows and pwsh/bash/zsh on Linux and macOS currently. I don't think it would be possible in command prompt and the git bash feature request is https://github.com/microsoft/vscode/issues/143769
